test/system/src/test/java/io/pravega/test/system/AbstractFailoverTests.java

```diff
@@ -374,6 +467,27 @@ void validateResults(ReaderGroupManager readerGroupManager, String readerGroupNa
         readerGroupManager.deleteReaderGroup(readerGroupName);
     }
 
+    void waitForScaling(String scope) throws InterruptedException, ExecutionException {
+        for (int waitCounter = 0; waitCounter < 2; waitCounter++) {
+            StreamSegments streamSegments = controller.getCurrentSegments(scope, AUTO_SCALE_STREAM).get();
+            testState.currentNumOfSegments.set(streamSegments.getSegments().size());
+            if (testState.currentNumOfSegments.get() == 2) {
+                log.info("The current number of segments is equal to 2, ScaleOperation did not happen");
+                //Scaling operation did not happen, wait
+                Exceptions.handleInterrupted(() -> Thread.sleep(60000));
```

Comment:
Rather than sleeping 60s and doing up to 2 iterations, why not sleep for say 10s and do up to 12 iterations. This will allow the test to run faster if the scaling is done before 60s.

Reply:
updated the code to use to wait for 12s in 12 iterations.